node/src/lib.rs

```diff
@@ -1,245 +1,99 @@
-pub enum StateTransition {
-    AddValidator {
-        /// The public key of the validator.
-        public_key: PublicKey,
-        /// The weight of the validator.
-        voting_power: VotingPower,
-    },
-    RemoveValidator {
-        /// The public key of the validator.
-        public_key: PublicKey,
-    },
-    Delegate {
-        /// The public key of the validator who delegates its voting right.
-        delegator: PublicKey,
-        /// The public key of the validator who is being delegated.
-        delegatee: PublicKey,
-        /// The target height of the block of this transaction, which is for preventing the replay attack.
-        target_height: BlockHeight,
-        /// The signature of (delegatee, target_height) signed by the delegator.
-        ///
-        /// This is a (non-essential) safety guard that assures that the delegator wants this action.
-        /// Otherwise (if this field doesn't exist), any sincere block proposer would include the commitment proof
-        /// in the transaction payload **anyway**, to convince other honest validators.
-        ///
-        /// That's why we just provide a reserved field for that.
-        commitment_proof: TypedSignature<(PublicKey, BlockHeight)>,
-    },
-    Undelegate {
-        /// The public key of the validator who claims its voting right.
-        delegator: PublicKey,
-    },
-    /// Inserts or updates (if exists) an arbitrary data entry.
-    InsertOrUpdateData { key: String, value: Vec<u8> },
-    /// Removes an existing data entry.
-    RemoveData(String),
-    /// Updates the protocol version
-    UpdateVersion { version: String },
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct NetworkStatus {
+    // TODO
 }
 
-/// A unit of state transition of the blockchain.
+/// The API for the Simperby node.
 ///
-/// Unlike other "ordinary" blockchains, Simperby doesn't have a signer nor a signature.
-/// TODO: Explain why.
-#[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone)]
-pub struct Transaction {
-    /// The instruction to perform on the blockchain state.
-    pub state_transition: Option<StateTransition>,
-    /// An optional field to store arbitrary data,
-    /// which is not part of the state but still useful as it's still part of the finalizing blockchain data
-    /// and so can be verified with the Merkle root.
-    ///
-    /// Note that it must be `Some` if the `state_transition` is `None`.
-    /// Otherwise, the transaction would be meaningless.
-    pub payload: Option<String>,
-}
+/// It is for serving the **CLI**, providing low-level funcitons and type-specified interfaces.
+#[async_trait]
+pub trait SimperbyApi<RR: RawRepository, R: DistributedRepository<RR>, C: Consensus, G: Governance>
+{
+    /// Initializes a new Simperby node from the genesis state
+    /// stored in the given directory (which is not yet a Git repository).
+    async fn genesis(&self) -> Result<()>;
```

Comment:
Will you use `Config::repository_directory` as the directory for genesis info?

Reply:
Yes, which is implied in the comment `which is not yet a Git repository`

Reply:
Ok